packages/workflow/src/internals.ts

```diff
@@ -651,8 +676,16 @@ export class Activator implements ActivationHandler {
         this.rejectUpdate(protocolInstanceId, error);
         return;
       }
+
+      const entry = this.updateHandlers.get(name);
+      if (!entry) {
+        return Promise.reject(new IllegalStateError(`No registered update handler for update: ${name}`));
+      }
+      const { unfinishedPolicy } = entry;
+
       const execute = composeInterceptors(this.interceptors.inbound, 'handleUpdate', this.updateNextHandler.bind(this));
       this.acceptUpdate(protocolInstanceId);
+      this.inProgressUpdates.set(updateId, { name, unfinishedPolicy, id: updateId });
       const res = execute(input)
         .then((result) => this.completeUpdate(protocolInstanceId, result))
         .catch((error) => {
```

Comment:
@mjameswh @bergundy is there a reason we catch `TemporalFailure` rather than `ApplicationFailure` here? (The test suite passes with either.)

Reply:
Hmm... we should reject the update in case of any error AFAICT, not just Temporal or Application failures.

Reply:
To answer my own question:

> is there a reason we catch `TemporalFailure` rather than `ApplicationFailure` here?

Yes, `Cancelledfailure` extends `TemporalFailure`, and we want workflow cancellation to result in rejection of the update, in the same way as workflow failure.

 > Hmm... we should reject the update in case of any error AFAICT, not just Temporal or Application failures.
 
There would be no point in rejecting the update in this case: the WFT is going to fail, so any messages/commands emit by the workflow will be dropped.